src/modules/change-state.js

```diff
@@ -178,19 +178,22 @@
 		}
 		
 		var followers = followerScope.find(followerSelector);
+		var notifyData = {item: item, state: state, flag: flag, cancel: false};
```

Comment:
This is dangerous because if one listener does

```js
var changeStateBegin = function (key, notifyData) {
    notifyData.cancel = someFunctionThatDoesLogicAndReturnsABoolean();
}
```

We might somehow overwrite the cancellation made by another listener.

Do work ok, we would always need to do

```js
var changeStateBegin = function (key, notifyData) {
    notifyData.cancel = notifyData.cancel || someFunctionThatDoesLogicAndReturnsABoolean();
}
```

which is prone to errors (because we forget).

If we want the cancel state to be un-undoable, we might need something like this

```js
var cancelled = false;
var notifyData = {item: item, state: state, flag: flag,
    cancel: function () { cancelled = true; }
};
```

That way, once it's called, no action can undo it.

Or we could have both the function and the flag
```js
var notifyData = {item: item, state: state, flag: flag, cancelled: false,
    cancel: function () { notifyData.cancelled = true; }
};
```

Reply:
@PascalPiche ping!